### PR TITLE
Update version to 12.3.3

### PR DESCRIPTION
**Pull request by eldada**
Updated some resources and documentation files with the lates 12.3.3 version.

### CODE REVIEW

File: pkg/artifactory/provider/provider.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
 package provider
 
-var Version = "12.0.0" // needs to be exported so make file can update this
+var Version = "12.3.3" // needs to be exported so make file can update this
```
Comment (alexhung) [inline pkg/artifactory/provider/provider.go:3]: This is automatically updated during build process by GoReleaser https://github.com/jfrog/terraform-provider-artifactory/blob/master/.goreleaser.yml#L16. So this is not necessary.

File: pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go [hunk 2]
```diff
@@ -1211,7 +1210,7 @@ func TestAccRemoteRepository_generic_migrate_to_schema_v4(t *testing.T) {
 				ExternalProviders: map[string]resource.ExternalProvider{
 					"artifactory": {
 						Source:            "jfrog/artifactory",
-						VersionConstraint: "12.0.0",
+						VersionConstraint: "12.3.3",
```
Comment (alexhung) [inline pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go:1213]: This is a TF schema migration test so the version constraint is specific to `12.0.0` (the previous release version with schema v3). See PR: https://github.com/jfrog/terraform-provider-artifactory/pull/1085/files

**Review by alexhung — CHANGES_REQUESTED** [review]
Couple of the updates are not necessary.

File: pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go [hunk 3]
```diff
@@ -10,7 +10,6 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/go-resty/resty/v2"
```
Comment (alexhung) [inline pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go:13]: Is this building for you? There's a reference to `*resty.Request` at https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go#L971

Comment (alexhung) [reply pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go:13]: https://github.com/jfrog/terraform-provider-artifactory/actions/runs/11669346571/job/32491100933#step:13:946

**Review by alexhung — CHANGES_REQUESTED** [review]
Build failed due to removal of `resty` import

File: pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go (hunk 3 above)
Comment (eldada) [reply pkg/artifactory/resource/repository/remote/resource_artifactory_remote_repository_test.go]: I put it back. It was auto removed in my ide

**Review by alexhung — APPROVED** [review]

**Pull request merged**